src/RemoteFileAPI/Remote.ts

```diff
@@ -24,24 +24,40 @@ export class Remote {
     this.connection?.close(eventCodeWhenIntentionallyStoppingConnection);
   }
 
-  public startConnection(): void {
+  public startConnection(autoConnectAttempt = 0): void {
```

Comment:
```suggestion
  public startConnection(autoConnectAttempt = 1): void {
```
I think defaulting to 1 will fix the "double message on startup" that I talked about.

Reply:
I made this a number specifically to get the double startup message. If that is not the wanted behaviour autoConnectAttempt should be a isAutoConnect boolean instead. The only ever check made against it is <= 1 which in your suggestion would make every connection attempt be classified as an auto-connect

EDIT: just saw the other suggestion so keeping it a number would be better after all

Reply:
Ah, that's a good point and it would be simpler that way.

To me, at least, the double startup error message feels like an odd exception. It feels more natural to only have one error shown between connects, regardless of how you get there.